### PR TITLE
feat: Add support for new select menus components

### LINKED ISSUE #61: Implement new Select Menus Components
Reference: https://discord.com/developers/docs/interactions/message-components#select-menus

### PR DESCRIPTION
**Pull request by Rapougnac**
# Description
Implements new options for select menus.

Resolves #61

## Connected issues & potential other potential problems
- https://github.com/nyxx-discord/nyxx/pull/380

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my changes haven't lowered code coverage

### CODE REVIEW

**Review by abitofevrything — DISMISSED** [review]

**Comment by Rapougnac** [comment]
As a question, should we cache reslut of `resolved` entries?

**Comment by Rapougnac** [comment]
JIC the received channels/users/members/roles from the payload are not in the cache

**Comment by l7ssha** [comment]
ah you mean that. Yeah sure, maybe we should do that. It should respect CacheOptions thing we have

File: lib/src/internal/utils.dart [hunk 1]
```diff
@@ -8,7 +8,7 @@ import 'package:nyxx_interactions/src/models/interaction.dart';
 import 'package:nyxx_interactions/src/models/slash_command.dart';
 
 /// Slash command names and subcommands names have to match this regex
-final RegExp slashCommandNameRegex = RegExp(r"^[\w-]{1,32}$");
+final RegExp slashCommandNameRegex = RegExp(r"^[-_\p{L}\p{N}\p{sc=Deva}\p{sc=Thai}]{1,32}$", unicode: true);
```
Comment (l7ssha) [inline lib/src/internal/utils.dart:11]: why such change?

File: lib/src/models/interaction.dart [hunk 2]
```diff
@@ -275,3 +274,86 @@ class MultiselectInteraction extends ComponentInteraction implements IMultiselec
...
+
+class MentionableMultiSelectInteraction extends ResolvedSelectInteraction implements IMentionableMultiSelectInteraction {
+  @override
+  late final Iterable<Mentionable> mentionables;
+
+  MentionableMultiSelectInteraction(INyxx client, RawApiMap raw) : super(client, raw) {
+    final resolved = InteractionDataResolved(raw['data']['resolved'] as RawApiMap, guild?.id, client);
+    mentionables = [
+      ...{...resolved.users, ...resolved.members},
+      ...resolved.roles
```
Comment (l7ssha) [inline lib/src/models/interaction.dart:341]: coudnt we just do `...resolved.users, ...resolved.members, ...resolved.roles` ?

**Review by l7ssha — CHANGES_REQUESTED** [review]

File: lib/src/internal/utils.dart (hunk 1 above)
Comment (abitofevrything) [reply lib/src/internal/utils.dart:11]: Discord has updated the regex. _commands updated it a while ago, but _interactions is still stuck on the old one...

File: lib/src/models/interaction.dart (hunk 2 above)
Comment (Rapougnac) [reply lib/src/models/interaction.dart:341]: Yeah

**Review by l7ssha — APPROVED** [review]
Looks good. Will be merged after release of nyxx

**Pull request merged**